private/bufpkg/bufconnect/interceptors_test.go

```diff
@@ -25,14 +25,32 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func TestAuthorizationInterceptorProviderTokenErr(t *testing.T) {
-	_, err := NewAuthorizationInterceptorProvider(app.NewEnvContainer(map[string]string{
-		tokenEnvKey: "test1234",
-	}))("fake")(func(context.Context, connect.AnyRequest) (connect.AnyResponse, error) {
-		return nil, errors.New("underlying cause")
+func TestNewAuthorizationInterceptorProvider(t *testing.T) {
+	tokenSet, err := NewTokenSetFromString("user1:token1@remote1,token")
+	assert.NoError(t, err)
+	_, err = NewAuthorizationInterceptorProvider(tokenSet)("default")(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		if req.Header().Get(AuthenticationHeader) != AuthenticationTokenPrefix+"token" {
+			return nil, errors.New("error auth token")
+		}
+		return nil, nil
 	})(context.Background(), connect.NewRequest(&bytes.Buffer{}))
+	assert.NoError(t, err)
+}
+
+func TestNewTokenSetFromEnv(t *testing.T) {
+	_, err := NewTokenSetFromContainer(app.NewEnvContainer(map[string]string{}))
+	assert.NoError(t, err)
+}
 
-	authErr, ok := AsAuthError(err)
-	assert.True(t, ok)
-	assert.Equal(t, tokenEnvKey, authErr.tokenEnvKey)
+func TestNewTokenSetFromString(t *testing.T) {
+	_, err := NewTokenSetFromString("default1,default2")
+	assert.Error(t, err)
+	_, err = NewTokenSetFromString("invalid@invalid@invalid")
+	assert.Error(t, err)
+	_, err = NewTokenSetFromString("invalid:invalid:invalid@remote")
+	assert.Error(t, err)
+	_, err = NewTokenSetFromString("user1:token1@remote,user2:token2@remote")
+	assert.Error(t, err)
+	_, err = NewTokenSetFromString("")
+	assert.NoError(t, err)
```

Comment:
Yay, tests! Here's some things to cover:

 - Add a test for a valid single token.
 - When you have two default tokens, test for the expected error.
 - Same goes for when you have duplicates for the same remote.
 - Check for two different tokens and users for two different remotes.

Reply:
Added a test for checking valid single token and checking different remote. For `two default token` and `duplicate remote`, do you mean we should also check if it is giving the correct error message?

Reply:
Checking for the expected error is what I was thinking. You don't have to but it can help if you want confidence that you're exercising the right check for the right kind of validity.